openmicroscopy/openmicroscopy
# Spw grid view

# What this PR does

First PR towards support for new SPW 'grid' layout.
This introduces a few changes, but not "full support" of https://github.com/openmicroscopy/design/issues/52
- When a Well is selected, right panel shows Well details & annotations (instead of Image)
- Image thumbnails are shown for selected well(s) below plate.
- Click to select an Image - this image will be shown in right panel.
- Plate and Well Images can both be zoomed with sliders

This PR does not support a number of 'grid view' features (to come in future PRs)
- Spacial layout of fields within well
- Set number of columns in grid-view

![screen shot 2016-09-23 at 15 43 35](https://cloud.githubusercontent.com/assets/900055/18790450/b413f082-81a6-11e6-986b-c2e212245dd3.png)
# Testing this PR
- Test Well annotations (Wells have not previously been annotatable)
  - Add, remove etc all types of annotations on Well.
  - Batch select Wells and add/remove annotations.
  - Metadata and Preview panels should be disabled for Wells
- Test loading of thumbnails for selected wells:
  - Select wells by Click or multi-select by drag or shift-click
  - For each selected well, a row of images will be loaded in bottom pane
  - Bottom pane can be resized by drag handle
  - Plate and well Images can be zoomed with sliders in bottom-right of each panel
- Clicking a single Image in bottom panel will load that image in Right panel.
- Can select Images / Wells in turn.
# TODO for this PR:
- [x] Multi-select Images
- [x] Indicate selected images with border etc.
- [x] Preserve zoom sizes when switching between Plates
- [x] Refresh Well grid thumbnails ( 'Save To All' or just 'Save')
- [ ] Handle browsing Tag tree -> Well. What do we show in tree/centre/right panels?
- [x] Tests for `marshal_tagged` in tree.py to test `wells`
